src/OutputWriters/windowed_time_average.jl

```diff
@@ -83,18 +84,59 @@ JLD2OutputWriter scheduled on TimeInterval(4 days):
 """
 function AveragedTimeInterval(interval; window=interval, stride=1)
     window > interval && throw(ArgumentError("Averaging window $window is greater than the output interval $interval."))
-    return AveragedTimeInterval(Float64(interval), Float64(window), stride, 0.0, false)
+    return AveragedTimeInterval(Float64(interval), Float64(window), stride, 0.0, 0, false)
 end
 
+# function initialize_schedule!(sch::AveragedTimeInterval, first_actuation_time::Number)
+#     sch.first_actuation_time = first_actuation_time
+#     sch.actuations = 0
+#     return true
+# end
+
+# initialize_schedule!(sch::AveragedTimeInterval, model) = initialize_schedule!(sch, model.clock.time)
+
+
+function next_actuation_time(sch::AveragedTimeInterval)
+    t₀ = sch.first_actuation_time
+    N = sch.actuations
+    T = sch.interval
+    return t₀ + (N + 1) * T
+end
+
+function (sch::AveragedTimeInterval)(model)
+    t = model.clock.time
+    t★ = next_actuation_time(sch)
+
+    if t >= t★
+        if sch.actuations < typemax(Int)
+            sch.actuations += 1
+        else # re-initialize the schedule to t★
+            initialize!(sch, t★)
+        end
+        return true
+    else
+        return false
+    end
+end
+
+# function schedule_aligned_time_step(sch::AveragedTimeInterval, clock, Δt)
+#     t★ = next_actuation_time(sch)
+#     t = clock.time
+#     return min(Δt, t★ - t)
+# end
+
 # Schedule actuation
-(sch::AveragedTimeInterval)(model) = sch.collecting || model.clock.time >= sch.previous_interval_stop_time + sch.interval - sch.window
-initialize_schedule!(sch::AveragedTimeInterval, clock) = sch.previous_interval_stop_time = clock.time - rem(clock.time, sch.interval)
-outside_window(sch::AveragedTimeInterval, clock) = clock.time <  sch.previous_interval_stop_time + sch.interval - sch.window   
-end_of_window(sch::AveragedTimeInterval, clock) = clock.time >= sch.previous_interval_stop_time + sch.interval
+# (sch::AveragedTimeInterval)(model) = sch.collecting || model.clock.time >= next_actuation_time(sch) - sch.window
+# initialize_schedule!(sch::AveragedTimeInterval, clock) = sch.previous_interval_stop_time = next_actuation_time(sch) - sch.interval
+initialize_schedule!(sch::AveragedTimeInterval, clock) = next_actuation_time(sch) - sch.interval
```

Comment:
I'm not 100% sure this is right, since it depends on `next_actuation_time` always being right.